testing/tests/test_e2e/test_secrets.py

```diff
@@ -505,12 +506,21 @@ def _on_secret_remove(self, event):
     ctx = Context(SecretCharm, meta={'name': 'foo'})
     secret = Secret({'a': 'b'}, owner='app')
     old_revision = 42
-    state = ctx.run(
-        ctx.on.secret_remove(secret, revision=old_revision),
-        State(leader=True, secrets={secret}),
-    )
+    if source == 'context':
```

Comment:
When there are parametrization *plus* large if/else blocks it also makes me wonder, "should this just be two separate test functions?"